static/app/utils/url/useLocationQuery.tsx

```diff
@@ -6,6 +6,12 @@ import {useLocation} from 'sentry/utils/useLocation';
 type Scalar = string | boolean | number | undefined;
 type Decoder = typeof decodeList | typeof decodeScalar | typeof decodeInteger;
 
+/**
+ * Select and memoize query params from location.
+ * This returns a new object only when one of your requested query params is
+ * updated, so you won't get re-renders whenever the overall location object
+ * changes.
```

Comment:
```suggestion
 * This returns a new object only when one of the specified query fields is
 * updated. The object will remain stable otherwise, avoiding re-renders.
```